volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,374 @@
+            requirements.BooleanRequirement(name="unix",
+                                            description=("Show UNIX domain Sockets only"),
+                                            default=False,
+                                            optional=True),
+            requirements.ListRequirement(name="pids",
+                                         description="Filter results by process IDs. "
+                                                     "It takes the root PID namespace identifiers.",
+                                         element_type=int,
+                                         optional=True),
+            requirements.IntRequirement(name="netns",
+                                        description="Filter results by network namespace. "
+                                                    "Otherwise, all of them are shown.",
+                                        optional=True),
+        ]
+
+    @classmethod
+    def list_sockets(cls,
+                     context: interfaces.context.ContextInterface,
+                     vmlinux_module_name: str,
+                     filter_func: Callable[[int], bool] = lambda _: False):
+        """
+        Returns every single socket descriptors
+        """
+        vmlinux = context.modules[vmlinux_module_name]
+
+        sfop_addr = vmlinux.object_from_symbol("socket_file_ops").vol.offset
+        dfop_addr = vmlinux.object_from_symbol("sockfs_dentry_operations").vol.offset
+
+        fd_generator = lsof.Lsof.list_fds(context, vmlinux.name, filter_func)
+        for _pid, _task_comm, task, fd_fields in fd_generator:
+            fd_num, filp, _full_path = fd_fields
+
+            if filp.f_op not in (sfop_addr, dfop_addr):
+                continue
+
+            dentry = filp.get_dentry()
+            if not dentry:
+                continue
+
+            d_inode = dentry.d_inode
+            if not d_inode:
+                continue
+
+            socket_alloc = linux.LinuxUtilities.container_of(d_inode, "socket_alloc", "vfs_inode", vmlinux)
+            _socket = socket_alloc.socket
+
+            vfs_inode = socket_alloc.vfs_inode
+            if not (_socket and vfs_inode):
+                continue
+
+            sock = _socket.sk.dereference()
+
+            sock_type = sock.type
+            family = sock.family
+
+            sock_handler = SockHandlers(vmlinux, task)
+            sock_fields = sock_handler.process_sock(sock)
+            if not sock_fields:
+                continue
+
+            child_sock = sock_fields[0]
+            protocol = child_sock.protocol if hasattr(child_sock, "protocol") else ""
+
+            net = task.nsproxy.net_ns
+            netns_id = net.proc_inum if net.has_member("proc_inum") else net.ns.inum
+            yield task, netns_id, fd_num, family, sock_type, protocol, sock_fields
+
+    def _generator(self):
+        pids = self.config.get('pids')
```

Comment:
Again, I'd prefer `self.config` access to be kept to `run` if possible.  Again, not a show stopper, just seems strange to create a function that isn't parameterized, but also isn't the top of the call chain...

There's a few more places further down, so if there are too many parameters, then it's ok, but ideally we'd try to avoid those (I don't know if there's other examples of generators calling the config, it's ok, but I'm trying to avoid it where possible, so not a rule, but guidelines where possible).  5:)